context.go

```diff
@@ -285,3 +286,43 @@ func normalizeFlags(flags []Flag, set *flag.FlagSet) error {
 	}
 	return nil
 }
+
+type requiredFlagsErr interface {
+	error
+	getMissingFlags() []string
```

Comment:
this function isn't used anywhere, but I was reading https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully and it said

> Assert errors for behaviour, not type

So that's what I'm trying to do here? 🤷‍♀ I'm open to suggestions for improvements!